app/com/linkedin/drelephant/spark/fetchers/statusapiv1/statusapiv1.scala

```diff
@@ -43,208 +43,418 @@ import scala.collection.Map
 
 import org.apache.spark.JobExecutionStatus
 import org.apache.spark.status.api.v1.StageStatus
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type
+import com.fasterxml.jackson.annotation.{JsonSubTypes, JsonTypeInfo}
 
-class ApplicationInfo(
-  val id: String,
-  val name: String,
-  val attempts: Seq[ApplicationAttemptInfo])
-
-class ApplicationAttemptInfo(
-  val attemptId: Option[String],
-  val startTime: Date,
-  val endTime: Date,
-  val sparkUser: String,
-  val completed: Boolean = false)
-
-class ExecutorStageSummary(
-  val taskTime : Long,
-  val failedTasks : Int,
-  val succeededTasks : Int,
-  val inputBytes : Long,
-  val outputBytes : Long,
-  val shuffleRead : Long,
-  val shuffleWrite : Long,
-  val memoryBytesSpilled : Long,
-  val diskBytesSpilled : Long)
-
-class ExecutorSummary(
-  val id: String,
-  val hostPort: String,
-  val rddBlocks: Int,
-  val memoryUsed: Long,
-  val diskUsed: Long,
-  val activeTasks: Int,
-  val failedTasks: Int,
-  val completedTasks: Int,
-  val totalTasks: Int,
-  val totalDuration: Long,
-  val totalInputBytes: Long,
-  val totalShuffleRead: Long,
-  val totalShuffleWrite: Long,
-  val maxMemory: Long,
-  val executorLogs: Map[String, String])
-
-class JobData(
-  val jobId: Int,
-  val name: String,
-  val description: Option[String],
-  val submissionTime: Option[Date],
-  val completionTime: Option[Date],
-  val stageIds: Seq[Int],
-  val jobGroup: Option[String],
-  val status: JobExecutionStatus,
-  val numTasks: Int,
-  val numActiveTasks: Int,
-  val numCompletedTasks: Int,
-  val numSkippedTasks: Int,
-  val numFailedTasks: Int,
-  val numActiveStages: Int,
-  val numCompletedStages: Int,
-  val numSkippedStages: Int,
-  val numFailedStages: Int)
+trait ApplicationInfo {
```

Comment:
@shankar37 
Why do we have these traits? Why can't we just have simple classes with `var` arguments?